### PR TITLE
Load tests

### PR DESCRIPTION
**Pull request by arturo-seijas**
Successful run: https://github.com/canonical/indico-operator/actions/runs/3929425567/jobs/6718345448

### CODE REVIEW

File: .github/workflows/integration_test.yaml [hunk 1]
```diff
@@ -253,6 +255,25 @@ jobs:
...
+      - name: Import load test json.env file
+        if: ${{ inputs.load-test-enabled }}
+        run: echo ${{ fromJSON(env.JSON_ENV) }}
+      - name: Reconfigure iptables
+        if: ${{ inputs.load-test-enabled }}
+        run: |
+          # Workaround to fix IP tables so that microk8s is reacheable on port 80 during the load test execution
+          # https://github.com/canonical/lxd-cloud/blob/f20a64a8af42485440dcbfd370faf14137d2f349/test/includes/lxd.sh#L13-L23
+          sudo iptables -P FORWARD ACCEPT
+          curl localhost:80
```
Comment (weiiwang01) [inline .github/workflows/integration_test.yaml]: What's the purpose of this command?

**Review by jdkandersson — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -19,23 +19,25 @@ The following workflows are available:
 
 * comment: Posts the content of the artifact specified as a comment in a PR. It needs to be triggered from a PR triggered workflow.
 
-* integration_test: Builds the existing Dockerfiles, if any, and executes the `integration` test target defined in the `tox.ini` file. The following parameters are available for this workflow:
+* integration_test: Builds the existing Dockerfiles, if any, and executes the `integration` test target defined in the `tox.ini` file. This workflow also supports running addtional load and charos tests. The following parameters are available for this workflow:
```
Comment (nrobinaubertin) [inline README.md:22]: s/charos/chaos

File: .github/workflows/integration_test.yaml [hunk 3]
```diff
@@ -75,6 +47,42 @@ on:
...
+          Each mapping will be injected into the matrix section of the integration-test.
+        type: string
+        default: '{}'
+      load-test-enabled:
+        type: boolean
+        description: Whether load testing is enabled
+        default: false
+      load-test-run-args:
+        type: string
+        description: Commend line arguments for the load test execution
```
Comment (nrobinaubertin) [inline .github/workflows/integration_test.yaml:65]: s/Commend/Command

**Review by nrobinaubertin — CHANGES_REQUESTED** [review]

File: .github/workflows/integration_test.yaml [hunk 4]
```diff
@@ -307,6 +314,12 @@ jobs:
           target: ${{ inputs.zap-target-protocol }}://${{ env.ZAP_TARGET }}:${{ inputs.zap-target-port }}/
           cmd_options: ${{ inputs.zap-cmd-options }}
           rules_file_name: ${{ inputs.zap-rules-file-name }}
+      - name: Install k6s
+        if: ${{ inputs.load-test-enabled }}
+        run: sudo snap install k6
```
Comment (gregory-schiano) [inline .github/workflows/integration_test.yaml:319]: Wasn't it a classic snap ?

Comment (arturo-seijas) [reply .github/workflows/integration_test.yaml:319]: No

File: .github/workflows/integration_test.yaml (hunk 1 above)
Comment (arturo-seijas) [reply .github/workflows/integration_test.yaml]: I was testing some things. Those lines have been removed.

**Review by weiiwang01 — DISMISSED** [review]

File: README.md [hunk 5]
```diff
@@ -19,23 +19,25 @@ The following workflows are available:
...
-| setup-devstack-swift | bool | false | Use setup-devstack-swift action to prepare a swift server for testing. |
 | chaos-enabled  | bool | false | Whether Chaos testing is enabled |
 | chaos-experiments | string | "" | List of experiments to run |
 | chaos-namespace | string | testing | Namespace to install Litmus Chaos |
 | chaos-app-namespace | string | testing | Namespace of chaos tested application |
 | chaos-app-label | string | "" | Label for chaos selection |
 | chaos-app-kind | string | statefulset | Application kind |
 | chaos-duration | string | 60 | Duration of the chaos experiment |
+| extra-arguments | string | "" | Additional arguments to pass to the integration test execution |
+| load-test-enabled | bool | false | Whether load testing is enabled. |
```
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply README.md]: Done

**Review by gregory-schiano — APPROVED** [review]
LGTM

**Review by weiiwang01 — APPROVED** [review]

**Review by nrobinaubertin — APPROVED** [review]

**Pull request merged**